evalml/automl/automl_algorithm/default_algorithm.py

```diff
@@ -484,35 +489,40 @@ def _transform_parameters(self, pipeline, proposed_parameters):
         return parameters
 
     def _make_split_pipeline(self, estimator, pipeline_name=None):
-        numeric_pipeline_parameters = {
-            "Select Columns Transformer": {"columns": self._selected_cols}
-        }
-        numeric_pipeline = make_pipeline(
-            self.X,
-            self.y,
-            estimator,
-            self.problem_type,
-            sampler_name=self.sampler_name,
-            parameters=numeric_pipeline_parameters,
-            extra_components=[SelectColumns],
-            extra_components_position="before_estimator",
-            use_estimator=False if self._selected_cat_cols else True,
-        )
-
         if self._selected_cat_cols:
             self._split = True
+
+            if self._X_with_cat_cols is None or self._X_without_cat_cols is None:
+                self._X_without_cat_cols = self.X.drop(self._selected_cat_cols, axis=1)
```

Comment:
@jeremyliweishih I think we should use ww here to select the columns. I think this is causing `make_pipeline` to reinfer the types which could cause problems